File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/polybius/PolybiusKey.kt
package cryptobench.ciphers.polybius

data class PolybiusKey(val seedWord: String) {

    /** 25-letter square: seed word first, then the rest of the alphabet minus J. */
    fun square(): String {
        val sb = StringBuilder()
        for (c in seedWord.uppercase().replace("J", "I") + "ABCDEFGHIKLMNOPQRSTUVWXYZ") {
            if (c in 'A'..'Z' && c != 'J' && !sb.contains(c)) sb.append(c)
        }
        return sb.toString()
    }

    companion object {
        fun default(): PolybiusKey = PolybiusKey("HARBOR")
    }
}

File: fixtures/scale/dart/large/lib/ciphers/polybius/polybius_key.dart
class PolybiusKey {
  final String seedWord;

  PolybiusKey(this.seedWord);

  /// 25-letter square: seed word first, then the rest of the alphabet minus J.
  String square() {
    final sb = StringBuffer();
    final source = '${seedWord.toUpperCase().replaceAll('J', 'I')}ABCDEFGHIKLMNOPQRSTUVWXYZ';
    for (final c in source.split('')) {
      if (c != 'J' && !sb.toString().contains(c)) sb.write(c);
    }
    return sb.toString();
  }

  factory PolybiusKey.defaultKey() => PolybiusKey('HARBOR');
}
